api/condition_types.go

```diff
@@ -39,6 +39,8 @@ const (
 	SucceedConditionType ConditionType = "Succeed"
 	// BuiltConditionType describes the condition of a resource that needs to be build.
 	BuiltConditionType ConditionType = "Built"
+	// DeployedConditionType describes the condition of a resource that needs to be deployed.
+	DeployedConditionType ConditionType = "Deployed"
```

Comment:
I'd avoid adding one more condition to the status since it requires a whole new status handling in the controller reconciliation algorithm.

We do the reconciliation based on the status in the `CanReconcile`. By adding an additional condition type we will have then to consider it in every cycle to avoid missing a combination of status and not doing a reconciliation whatsoever for a given object.

Reply:
OK. It is removed.